### ISSUE TITLE
[v5 + BPCells] No layer matching pattern 'data' found. Please run NormalizeData and retry

### ISSUE THREAD

**Issue by menagerlab** [satijalab/seurat#8119]
When I run `IntegrateLayers` on a 100 samples object, I get the following error:

    > integrated_obj <- IntegrateLayers(integrated_obj, method = RPCAIntegration, normalization.method ="LogNormalize",
    +                                         group.by = "SampleID", orig.reduction = "pca", new.reduction = new_reduction,
    +                                         verbose = TRUE, reference = ref_samples_idx)
    Error in `ScaleData()`:
    ! No layer matching pattern 'data' found. Please run NormalizeData and retry
    Run `rlang::last_trace()` to see where the error occurred.

I did normalise and scale the object before attempting the integration, and the same piece of code was working in the beta version of `Seurat`.

This issue should be linked with both [#8004](https://github.com/satijalab/seurat/issues/8004) and [#7936](https://github.com/satijalab/seurat/issues/7936), but this case is slightly different as I am only working with `v5` objects and I am not trying to save.

This is how the object looks like before the integration step

   > integrated_obj[["RNA"]]
   Assay (v5) data with 32334 features for 653530 cells
   Top 10 variable features:
    JCHAIN, PPBP, IGKC, TOP2A, MZB1, HIST1H4C, CENPF, GP1BB, TPX2, KRT18 
   Layers:
   counts.C20-59-01-P005, counts.C20-59-01-P016, counts.C20-59-01-P017, counts.C20-59-01-P018, counts.C20-59-01-P024, counts.C20-59-04-C017, counts.C20-59-04-C015, counts.C20-59-04-C019, counts.C20-59-01-P035, counts.C20-59-01-P020, counts.C20-59-01-P023, counts.C20-59-01-P028, counts.C20-59-01-P029, counts.C20-59-04-C016, counts.C20-59-04-C014, counts.C20-59-04-C010, counts.C20-59-02-P001, counts.C20-59-02-P002, counts.C20-59-01-P010,       counts.C20-59-01-P022, counts.C20-59-01-P026, counts.C20-59-01-P027, counts.C20-59-04-C020, counts.C20-59-04-C021, counts.C20-59-02-C001, counts.C20-59-04-C013, counts.C20-59-04-C018, counts.C20-59-01-P025, counts.C20-59-01-P032, counts.C20-59-01-P033, counts.C20-59-01-P043, counts.C20-59-01-P047, counts.C20-59-04-C007, counts.C20-59-04-C022, counts.C20-59-01-P042, counts.C20-59-01-P041, counts.C20-59-01-P045, counts.C20-59-01-P049, counts.C20-59-01-P050, counts.C20-59-01-P053, counts.C20-59-01-C001, counts.C20-59-01-C002, counts.C20-59-01-P066, counts.C20-59-01-P067, counts.C20-59-01-P065, counts.C20-59-01-C003, counts.C20-59-04-C023, counts.C20-59-01-P075, counts.C20-59-01-P087, counts.C20-59-01-P052, counts.C20-59-01-P097, counts.C20-59-01-P069, counts.C20-59-07-C005, counts.C20-59-07-C001, counts.C20-59-01-P084, counts.C20-59-01-P093, counts.C20-59-01-P094, counts.C20-59-01-P096, counts.C20-59-01-P088, counts.C20-59-07-C006, counts.C20-59-07-C010, counts.C20-59-04-C024, counts.C20-59-01-P102, counts.C20-59-01-P103, counts.C20-59-07-C002, counts.C20-59-07-C003, counts.C20-59-05-C002, counts.C20-59-01-P074, counts.C20-59-01-P090, counts.C20-59-01-P107, counts.C20-59-07-C018, counts.C20-59-07-C020, counts.C20-59-01-P063, counts.C20-59-01-P072, counts.C20-59-01-P073, counts.C20-59-01-P076, counts.C20-59-01-P081, counts.C20-59-05-C001, counts.C20-59-07-C013, counts.C20-59-07-C019, counts.C20-59-01-P036, counts.C20-59-01-P071, counts.C20-59-01-P079, counts.C20-59-01-P080, counts.C20-59-01-P086, counts.C20-59-07-C028, counts.C20-59-05-C009, counts.C20-59-05-C010, counts.C20-59-01-P056, counts.C20-59-01-P068, counts.C20-59-01-P077, counts.C20-59-01-P085, counts.C20-59-01-P095, counts.C20-59-04-C026, counts.C20-59-07-C022, counts.C20-59-07-C029, data.C20-59-01-P005, data.C20-59-01-P016, data.C20-59-01-P017, data.C20-59-01-P018, data.C20-59-01-P024, data.C20-59-04-C017, data.C20-59-04-C015, data.C20-59-04-C019, data.C20-59-01-P035, data.C20-59-01-P020, data.C20-59-01-P023, data.C20-59-01-P028, data.C20-59-01-P029, data.C20-59-04-C016, data.C20-59-04-C014, data.C20-59-04-C010, data.C20-59-02-P001, data.C20-59-02-P002, data.C20-59-01-P010, data.C20-59-01-P022, data.C20-59-01-P026, data.C20-59-01-P027, data.C20-59-04-C020, data.C20-59-04-C021, data.C20-59-02-C001, data.C20-59-04-C013, data.C20-59-04-C018, data.C20-59-01-P025, data.C20-59-01-P032, data.C20-59-01-P033, data.C20-59-01-P043, data.C20-59-01-P047, data.C20-59-04-C007, data.C20-59-04-C022, data.C20-59-01-P042, data.C20-59-01-P041, data.C20-59-01-P045, data.C20-59-01-P049, data.C20-59-01-P050, data.C20-59-01-P053, data.C20-59-01-C001, data.C20-59-01-C002, data.C20-59-01-P066, data.C20-59-01-P067, data.C20-59-01-P065, data.C20-59-01-C003, data.C20-59-04-C023, data.C20-59-01-P075, data.C20-59-01-P087, data.C20-59-01-P052, data.C20-59-01-P097, data.C20-59-01-P069, data.C20-59-07-C005, data.C20-59-07-C001, data.C20-59-01-P084, data.C20-59-01-P093, data.C20-59-01-P094, data.C20-59-01-P096, data.C20-59-01-P088, data.C20-59-07-C006, data.C20-59-07-C010, data.C20-59-04-C024, data.C20-59-01-P102, data.C20-59-01-P103, data.C20-59-07-C002, data.C20-59-07-C003, data.C20-59-05-C002, data.C20-59-01-P074, data.C20-59-01-P090, data.C20-59-01-P107, data.C20-59-07-C018, data.C20-59-07-C020, data.C20-59-01-P063, data.C20-59-01-P072, data.C20-59-01-P073, data.C20-59-01-P076, data.C20-59-01-P081, data.C20-59-05-C001, data.C20-59-07-C013, data.C20-59-07-C019, data.C20-59-01-P036, data.C20-59-01-P071, data.C20-59-01-P079, data.C20-59-01-P080, data.C20-59-01-P086, data.C20-59-07-C028, data.C20-59-05-C009, data.C20-59-05-C010, data.C20-59-01-P056, data.C20-59-01-P068, data.C20-59-01-P077, data.C20-59-01-P085, data.C20-59-01-P095, data.C20-59-04-C026, data.C20-59-07-C022, data.C20-59-07-C029, scale.data
 1 dimensional reduction calculated: pca

This is the code I am using

    # sample_list: list containing the counts  (sample_list[[sample_name]] <- sample_name@assays$RNA$counts)
    # metadata_list: full metadata table
    integrated_obj <- CreateSeuratObject(counts = sample_list, meta.data = metadata_list)
   
    integrated_obj <- NormalizeData(integrated_obj, verbose = TRUE)
    integrated_obj <- FindVariableFeatures(integrated_obj, nfeatures = 3000, verbose = TRUE)
    integrated_obj <- ScaleData(integrated_obj, verbose = TRUE)
    integrated_obj <- RunPCA(integrated_obj, verbose = TRUE)
   
    # get index of samples to use as reference
    ref_samples <- c("C20-59-07-C022","C20-59-01-C002",..)
    ref_samples_idx <- which(Layers(integrated_obj) %in% paste0("data.",ref_samples))
  
    integrated_obj <- IntegrateLayers(integrated_obj, method = RPCAIntegration, normalization.method ="LogNormalize",
                                        group.by = "SampleID", orig.reduction = "pca", new.reduction = "red_rpca",
                                        verbose = TRUE, reference = ref_samples_idx)

At first I thought it was an issue with how I was passing the reference samples (`reference=ref_samples_idx`), but I get the same error without using a reference

    integrated_obj <- IntegrateLayers(integrated_obj, method = RPCAIntegration, normalization.method ="LogNormalize",
                                        group.by = "SampleID", orig.reduction = "pca", new.reduction = "red_rpca",
                                        verbose = TRUE)
    Error in `ScaleData()`:
    ! No layer matching pattern 'data' found. Please run NormalizeData and retry
    Run `rlang::last_trace()` to see where the error occurred.

`HarmonyIntegration` seems to work (20 min in and still no errors)

    >       integrated_obj <- IntegrateLayers(integrated_obj, method = HarmonyIntegration,
    +                                         group.by = "SampleID", orig.reduction = "pca", new.reduction = "red_harmony,
    +                                         verbose = TRUE, reference = ref_samples_idx)

    > sessionInfo()
    R version 4.3.2 (2023-10-31)
    Platform: x86_64-pc-linux-gnu (64-bit)
    Running under: Debian GNU/Linux 10 (buster)

    Matrix products: default
    BLAS:   /usr/lib/x86_64-linux-gnu/libf77blas.so.3.10.3 
    LAPACK: /usr/lib/x86_64-linux-gnu/atlas/liblapack.so.3.10.3;  LAPACK version 3.8.0

    locale:
     [1] LC_CTYPE=fr_FR.UTF-8       LC_NUMERIC=C               LC_TIME=fr_FR.UTF-8        LC_COLLATE=fr_FR.UTF-8     LC_MONETARY=fr_FR.UTF-8    LC_MESSAGES=fr_FR.UTF-8    LC_PAPER=fr_FR.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
    [10] LC_TELEPHONE=C             LC_MEASUREMENT=fr_FR.UTF-8 LC_IDENTIFICATION=C       

    time zone: Europe/Paris
    tzcode source: system (glibc)

    attached base packages:
    [1] stats     graphics  grDevices utils     datasets  methods   base     

    other attached packages:
     [1] future_1.33.0         sctransform_0.4.1     scR_0.9.3             Azimuth_0.5.0         shinyBS_0.61.1        SeuratDisk_0.0.0.9021 Seurat_5.0.1          SeuratObject_5.0.0    sp_2.1-1              BPCells_0.1.0         patchwork_1.1.3      
    [12] ggplot2_3.4.4        

    loaded via a namespace (and not attached):
      [1] fs_1.6.3                          ProtGenerics_1.34.0               matrixStats_1.1.0                 spatstat.sparse_3.0-3             bitops_1.0-7                              DirichletMultinomial_1.44.0       TFBSTools_1.40.0                 
      [8] httr_1.4.7                        RColorBrewer_1.1-3                tools_4.3.2                       utf8_1.2.4                        R6_2.5.1                                  DT_0.30                           lazyeval_0.2.2                   
     [15] uwot_0.1.16                       rhdf5filters_1.14.1               withr_2.5.2                       prettyunits_1.2.0                 gridExtra_2.3                     progressr_0.14.0                  cli_3.6.1                        
     [22] Biobase_2.62.0                    spatstat.explore_3.2-5            fastDummies_1.7.3                 EnsDb.Hsapiens.v86_2.99.0         shinyjs_2.1.0                     spatstat.data_3.0-3               readr_2.1.4                      
     [29] ggridges_0.5.4                    pbapply_1.7-2                     Rsamtools_2.18.0                  R.utils_2.12.2                    harmony_1.1.0                     parallelly_1.36.0                 BSgenome_1.70.1                  
     [36] RSQLite_2.3.3                     shape_1.4.6                       generics_0.1.3                    BiocIO_1.12.0                     gtools_3.9.4                      ica_1.0-3                         spatstat.random_3.2-1            
     [43] zip_2.3.0                         googlesheets4_1.1.1               dplyr_1.1.3                       GO.db_3.18.0                      Matrix_1.6-3                      fansi_1.0.5                       S4Vectors_0.40.1                 
     [50] abind_1.4-5                       R.methodsS3_1.8.2                 lifecycle_1.0.4                   yaml_2.3.7                        SummarizedExperiment_1.32.0       rhdf5_2.46.0                      SparseArray_1.2.2                
     [57] BiocFileCache_2.10.1              Rtsne_0.16                        grid_4.3.2                        blob_1.2.4                        promises_1.2.1                    shinydashboard_0.7.2              crayon_1.5.2                     
     [64] miniUI_0.1.1.1                    lattice_0.22-5                    cowplot_1.1.1                     GenomicFeatures_1.54.1            annotate_1.80.0                   KEGGREST_1.42.0                   pillar_1.9.0                     
     [71] GenomicRanges_1.54.1              rjson_0.2.21                      future.apply_1.11.0               codetools_0.2-19                  fastmatch_1.1-4                   leiden_0.4.3                      glue_1.6.2                       
     [78] data.table_1.14.8                 vctrs_0.6.4                       png_0.1-8                         spam_2.10-0                       cellranger_1.1.0                  gtable_0.3.4                      poweRlaw_0.70.6                  
     [85] cachem_1.0.8                      openxlsx_4.2.5.2                  Signac_1.12.0                     S4Arrays_1.2.0                    mime_0.12                         pracma_2.4.4                      survival_3.5-7                   
     [92] gargle_1.5.2                      RcppRoll_0.3.0                    ellipsis_0.3.2                    fitdistrplus_1.1-11               ROCR_1.0-11                       nlme_3.1-163                      bit64_4.0.5                      
     [99] progress_1.2.2                    filelock_1.0.2                    RcppAnnoy_0.0.21                  GenomeInfoDb_1.38.1               irlba_2.3.5.1                     KernSmooth_2.23-22                colorspace_2.1-0                 
    [106] seqLogo_1.68.0                    BiocGenerics_0.48.1               DBI_1.1.3                         tidyselect_1.2.0                  bit_4.0.5                         compiler_4.3.2                    curl_5.1.0                       
    [113] hdf5r_1.3.8                       xml2_1.3.5                        DelayedArray_0.28.0               plotly_4.10.3                     rtracklayer_1.62.0                scales_1.2.1                      caTools_1.18.2                   
    [120] lmtest_0.9-40                     rappdirs_0.3.3                    stringr_1.5.1                     digest_0.6.33                     goftest_1.2-3                     presto_1.0.0                      spatstat.utils_3.0-4             
    [127] XVector_0.42.0                    htmltools_0.5.7                   pkgconfig_2.0.3                   MatrixGenerics_1.14.0             dbplyr_2.4.0                      fastmap_1.1.1                     ensembldb_2.26.0                 
    [134] GlobalOptions_0.1.2               rlang_1.1.2                       htmlwidgets_1.6.2                 shiny_1.7.5.1                     zoo_1.8-12                        jsonlite_1.8.7                    BiocParallel_1.36.0              
    [141] R.oo_1.25.0                       RCurl_1.98-1.13                   magrittr_2.0.3                    GenomeInfoDbData_1.2.11           dotCall64_1.1-0                   Rhdf5lib_1.24.0                   munsell_0.5.0                    
    [148] Rcpp_1.0.11                       reticulate_1.34.0                 stringi_1.8.1                     zlibbioc_1.48.0                   MASS_7.3-60                       plyr_1.8.9                        parallel_4.3.2                   
    [155] listenv_0.9.0                     ggrepel_0.9.4                     deldir_1.0-9                      CNEr_1.38.0                       Biostrings_2.70.1                 splines_4.3.2                     tensor_1.5                       
    [162] circlize_0.4.15                   hms_1.1.3                         BSgenome.Hsapiens.UCSC.hg38_1.4.5 igraph_1.5.1                      spatstat.geom_3.2-7               RcppHNSW_0.5.0                    reshape2_1.4.4                   
    [169] biomaRt_2.58.0                    stats4_4.3.2                      TFMPvalue_0.0.9                   XML_3.99-0.15                     JASPAR2020_0.99.10                tzdb_0.4.0                        httpuv_1.6.12                    
    [176] RANN_2.6.1                        tidyr_1.3.0                       purrr_1.0.2                       polyclip_1.10-6                   SeuratData_0.2.2.9001             scattermore_1.2                   xtable_1.8-4                     
    [183] restfulr_0.0.15                   AnnotationFilter_1.26.0           RSpectra_0.16-1                   later_1.3.1                       googledrive_2.1.1                 viridisLite_0.4.2                 tibble_3.2.1                     
    [190] memoise_2.0.1                     AnnotationDbi_1.64.1              GenomicAlignments_1.38.0          IRanges_2.36.0                    cluster_2.1.4                     globals_0.16.2

**Comment by lisch7** [comment]
Same issue.

**Comment by maartenciers** [comment]
Same issue.

**Comment by Gesmira** [comment]
Hi, 
Are you able to reproduce this with pbmc3k or share a reproducible and small version of the object with the issue?

**Comment by SkrHo** [comment]
Same issue after updating Seurat.

**Comment by Gesmira** [comment]
Hi @jlmalin, 
Sorry for the late response, you can send me the object at gmolla@nygenome.org.

**Comment by redwanfarooq** [comment]
Hello, just wondering whether there has been any progress in addressing this issue? I'm having the same problem when trying to run `IntegrateLayers` on a Seurat v5 object using BPCells on-disk matrices. Here's a small reproducible example using the 10X pbmc4k and pbmc8k datasets downloaded from [here](https://support.10xgenomics.com/single-cell-gene-expression/datasets/2.1.0/pbmc4k) and [here](https://support.10xgenomics.com/single-cell-gene-expression/datasets/2.1.0/pbmc8k). The issue seems to be specific to using BPCells on-disk matrices, as the code snippet below runs without any errors if I just use dgCMatrix class matrices to create a Seurat object by omitting the section commented as "convert to BPCells on-disk matrices". Thanks in advance for your help!

**Code:**
```
library(dplyr)
library(glue)
library(Seurat)
library(BPCells)

pbmc4k <- Read10X("pbmc4k/filtered_gene_bc_matrices/GRCh38")
pbmc8k <- Read10X("pbmc8k/filtered_gene_bc_matrices/GRCh38")

counts <- list("pbmc4k" = pbmc4k, "pbmc8k" = pbmc8k)
# convert to BPCells on-disk matrices ----------------------
counts <- Map(
  function(x, dataset) {
    write_matrix_dir(
      mat = convert_matrix_type(x, type = "uint32_t"),
      dir = glue("{dataset}/matrix_dir"),
      overwrite = TRUE
    )
  },
  x = counts,
  dataset = names(counts)
)
# ---------------------------------------------------------

seu <- Map(
  function(x, dataset) {
    CreateSeuratObject(
      counts = x,
      assay = "RNA",
      project = dataset
    )
  },
  x = counts,
  dataset = names(counts)
)
seu <- merge(x = seu$pbmc4k, y = seu$pbmc8k, add.cell.ids = names(seu))
seu <- seu %>%
  NormalizeData() %>%
  FindVariableFeatures() %>%
  ScaleData() %>%
  RunPCA() %>%
  IntegrateLayers(method = RPCAIntegration, orig.reduction = "pca", new.reduction = "integrated")
```

**Error message:**
```
Error in `ScaleData()`:
! No layer matching pattern 'data' found. Please run NormalizeData and retry
Backtrace:
     ▆
  1. ├─... %>% ...
  2. └─Seurat::IntegrateLayers(...)
  3.   └─Seurat (local) method(...)
  4.     ├─base::suppressWarnings(ScaleData(object = object.list[[i]], verbose = FALSE))
  5.     │ └─base::withCallingHandlers(...)
  6.     ├─Seurat::ScaleData(object = object.list[[i]], verbose = FALSE)
  7.     └─Seurat:::ScaleData.Seurat(object = object.list[[i]], verbose = FALSE)
  8.       ├─Seurat::ScaleData(...)
  9.       └─Seurat:::ScaleData.StdAssay(...)
 10.         └─rlang::abort(paste0("No layer matching pattern '", olayer, "' found. Please run NormalizeData and retry"))
```

**Session info:**
```
R version 4.3.2 (2023-10-31)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: Ubuntu 22.04.2 LTS

Matrix products: default
BLAS/LAPACK: [/ceph/project/fuggerlab/rfarooq/.conda/envs/sc-sandbox/lib/libopenblasp-r0.3.25.so](https://vscode-remote+ssh-002dremote-002bjade2.vscode-resource.vscode-cdn.net/ceph/project/fuggerlab/rfarooq/.conda/envs/sc-sandbox/lib/libopenblasp-r0.3.25.so);  LAPACK version 3.11.0

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
 [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/London
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] BPCells_0.1.0      Seurat_5.0.1       SeuratObject_5.0.1 sp_2.1-2          
[5] glue_1.7.0         dplyr_1.1.4       

loaded via a namespace (and not attached):
  [1] deldir_2.0-2           pbapply_1.7-2          gridExtra_2.3         
  [4] rlang_1.1.3            magrittr_2.0.3         RcppAnnoy_0.0.21      
  [7] spatstat.geom_3.2-7    matrixStats_1.2.0      ggridges_0.5.5        
 [10] compiler_4.3.2         png_0.1-8              vctrs_0.6.5           
 [13] reshape2_1.4.4         stringr_1.5.1          pkgconfig_2.0.3       
 [16] crayon_1.5.2           fastmap_1.1.1          ellipsis_0.3.2        
 [19] utf8_1.2.4             promises_1.2.1         purrr_1.0.2           
 [22] jsonlite_1.8.8         goftest_1.2-3          later_1.3.2           
 [25] uuid_1.2-0             spatstat.utils_3.0-4   irlba_2.3.5.1         
 [28] parallel_4.3.2         cluster_2.1.6          R6_2.5.1              
 [31] ica_1.0-3              spatstat.data_3.0-4    stringi_1.8.3         
 [34] RColorBrewer_1.1-3     reticulate_1.34.0      parallelly_1.36.0     
 [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.12           
 [40] IRkernel_1.3.2         tensor_1.5             future.apply_1.11.1   
 [43] zoo_1.8-12             base64enc_0.1-3        sctransform_0.4.1     
 [46] httpuv_1.6.13          Matrix_1.6-3           splines_4.3.2         
 [49] igraph_1.5.1           tidyselect_1.2.0       abind_1.4-5           
 [52] spatstat.random_3.2-2  codetools_0.2-19       miniUI_0.1.1.1        
 [55] spatstat.explore_3.2-5 listenv_0.9.0          lattice_0.22-5        
 [58] tibble_3.2.1           plyr_1.8.9             shiny_1.8.0           
 [61] ROCR_1.0-11            evaluate_0.23          Rtsne_0.17            
 [64] future_1.33.1          fastDummies_1.7.3      survival_3.5-7        
 [67] polyclip_1.10-6        fitdistrplus_1.1-11    pillar_1.9.0          
 [70] KernSmooth_2.23-22     plotly_4.10.4          generics_0.1.3        
 [73] RcppHNSW_0.5.0         IRdisplay_1.1          ggplot2_3.4.4         
 [76] munsell_0.5.0          scales_1.3.0           globals_0.16.2        
 [79] xtable_1.8-4           lazyeval_0.2.2         tools_4.3.2           
 [82] data.table_1.14.10     RSpectra_0.16-1        pbdZMQ_0.3-11         
 [85] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       
 [88] cowplot_1.1.2          grid_4.3.2             tidyr_1.3.0           
 [91] colorspace_2.1-0       nlme_3.1-164           patchwork_1.2.0       
 [94] repr_1.1.6             cli_3.6.2              spatstat.sparse_3.0-3 
 [97] spam_2.10-0            fansi_1.0.6            viridisLite_0.4.2     
[100] uwot_0.1.16            gtable_0.3.4           digest_0.6.34         
[103] progressr_0.14.0       ggrepel_0.9.5          htmlwidgets_1.6.4     
[106] htmltools_0.5.7        lifecycle_1.0.4        httr_1.4.7            
[109] mime_0.12              MASS_7.3-60
```

**Comment by Gesmira** [comment]
Hi, 
We have solved the issue and it will be included in the `Seurat 5.0.2` release coming out shortly. I will update here as well when that is out.
